File: showdown/internal/domain/hand.go
package domain

import "fmt"

type Hand struct {
	cards []Card
}

func (h *Hand) AddCard(c Card) {
	h.cards = append(h.cards, c)
}

func (h *Hand) PickCard(order int) Card {
	c := h.cards[order]
	h.cards = append(h.cards[:order], h.cards[order+1:]...)
	return c
}

func (h *Hand) Size() int {
	return len(h.cards)
}

func (h *Hand) List() {
	for i, c := range h.cards {
		fmt.Printf("%d: %s\n", i, c.String())
	}
}

File: 1_1_cpp/domain/hand.cpp
#include <vector>
#include <iostream>
#include "hand.h"
#include "card.h"


void Hand::AddCard(const Card& c) {
    cards.push_back(c);
}

Card Hand::PickCard(int order) {
    Card c = cards[order];
    cards.erase(cards.begin() + order);
    return c;
}

int Hand::Size() const {
    return static_cast<int>(cards.size());
}

void Hand::List() const {
    for (int i = 0; i < static_cast<int>(cards.size()); ++i) {
        Card c = cards[i];
        std::cout << i << ": " << c.String() << std::endl;
    }
}